File: include/PolyWorksSDK/samples/plugins/common/VisualBasic.NET/VBNetCommands/COM_SDK/Util/CComCommandParameterEnumImpl.vb
' =================================================================================================
' Implementation of the class for COM command parameters enumeration
'
' Copyright © InnovMetric Logiciels Inc. 2011 All Rights Reserved
' -------------------------------------------------------------------------------------------------
Namespace VBNetCommands

    Public Class CComCommandParameterEnumImpl

        ' =============================================================================
        '                           IIMCommandParameterEnum Interface
        ' =============================================================================

        ' =============================================================================
        ' See CComCommandParameterEnum.ParamsGetNb
        ' -----------------------------------------------------------------------------
        Public Function ParamsGetNb() As Integer

            If Not m_parameters Is Nothing Then

                Return m_parameters.Count

            End If

            Return 0

        End Function


        ' =============================================================================
        ' See CComCommandParameterEnum.ParamGet
        ' -----------------------------------------------------------------------------
        Public Sub ParamGet(ByVal number_ As Integer, ByRef pIIMCommandParameter_ As IMPluginLib.IIMCommandParameter)

            pIIMCommandParameter_ = Nothing

            If m_parameters Is Nothing Then

                Return

            End If

            If number_ >= 0 And number_ < m_parameters.Count Then

                pIIMCommandParameter_ = m_parameters(number_)

            End If

        End Sub


        ' =============================================================================
        '               CComCommandParameterEnumImpl specific methods
        ' =============================================================================

        ' =============================================================================
        ' Add an argument to the enumeration
        '
        ' Parameter: pCommandParameter_ : Argument to add
        ' -----------------------------------------------------------------------------
        Public Sub CommandParameterAdd(ByVal commandParameter_ As IMPluginLib.IIMCommandParameter)

            If Not m_parameters Is Nothing Then

                m_parameters.Add(commandParameter_)

            End If

        End Sub

        Private m_parameters As System.Collections.Generic.List(Of IMPluginLib.IIMCommandParameter) = New System.Collections.Generic.List(Of IMPluginLib.IIMCommandParameter)()

    End Class

End Namespace

File: include/PolyWorksSDK/samples/plugins/common/CSharp/CSharpCommands/COM_SDK/Util/CComCommandParameterEnumImpl.cs
// =================================================================================================
// Implementation of the class for COM command parameters enumeration
//
// Copyright © InnovMetric Logiciels Inc. 2009 All Rights Reserved
// -------------------------------------------------------------------------------------------------
namespace IM
{    
    public class CComCommandParameterEnumImpl
    {
        /* *****************************************************************************
        *                           IIMCommandParameterEnum Interface
        * -------------------------------------------------------------------------- */
        #region IIMCommandParameterEnum

        // =============================================================================
        // See CComCommandParameterEnum.ParamsGetNb
        // -----------------------------------------------------------------------------
        public int ParamsGetNb()
        {
            if(m_parameters != null)
            {
                return m_parameters.Count;
            }
            return 0;
        }

        // =============================================================================
        // See CComCommandParameterEnum.ParamGet
        // -----------------------------------------------------------------------------
        public void ParamGet(int number_, out IMPluginLib.IIMCommandParameter iIMCommandParameter_)
        {
            iIMCommandParameter_ = null;

            if( m_parameters == null)
            {
                return;
            }

            if( number_ >= 0 && number_ < m_parameters.Count)
            {
                iIMCommandParameter_ = m_parameters[number_];
            }
        }

        #endregion

        /* *****************************************************************************
         *              CComCommandParameterEnumImpl specific methods
        * -------------------------------------------------------------------------- */
        #region CComCommandParameterEnumImpl

        // =============================================================================
        // Add an argument to the enumeration
        //
        // Parameter: pCommandParameter_ : Argument to add
        // -----------------------------------------------------------------------------
        public void CommandParameterAdd(IMPluginLib.IIMCommandParameter commandParameter_)
        {
            if (m_parameters != null)
            {
                m_parameters.Add(commandParameter_);
            }
        }

        private System.Collections.Generic.List<IMPluginLib.IIMCommandParameter> m_parameters = new System.Collections.Generic.List<IMPluginLib.IIMCommandParameter>();

        #endregion
    }
}
